### PR TITLE
an e2e test for checking scaling up by one master node

### PR DESCRIPTION
**Pull request by p0lyn0mial**
a simple e2e test that adds a new master node and checks if it will result in an increase in etcd cluster

### CODE REVIEW

File: test/e2e/vertical_scaling_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	etcdPods, err := cs.CoreV1Interface.Pods("").List(context.TODO(), metav1.ListOptions{LabelSelector: "k8s-app=etcd"})
+	require.NoError(t, err)
+	etcdPodName := etcdPods.Items[0].Name
+
+	waitPollInterval := 15 * time.Second
+	waitPollTimeout := 5 * time.Minute
+	t.Logf("Waiting up to %s for a new etcd member to join the cluster", waitPollTimeout.String())
+
+	if err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		ocCmd := strings.Split(fmt.Sprintf("rsh -c etcdctl -n openshift-etcd %s %s", etcdPodName, "etcdctl member list -w=json"), " ")
```
Comment (p0lyn0mial) [inline test/e2e/vertical_scaling_test.go:134]: I am using `oc` for getting the member list, in the future, we could consume etcd API. I think it is a good starting point.

File: test/e2e/vertical_scaling_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	for _, machine := range machineList.Items {
+		machinePhase := pointer.StringDeref(machine.Status.Phase, "")
+		if machinePhase != "Running" {
+			t.Fatalf("%q machine is in unexpected %q state", machine.Name, machinePhase)
+		}
+	}
+}
+
+// ensureMembersCount simply counts the current etcd members, it doesn't evaluate health conditions or any other attributes (i.e. name) of individual members
+func ensureMembersCount(t testing.TB, cs *framework.ClientSet, expectedMembersCount int) {
```
Comment (p0lyn0mial) [inline test/e2e/vertical_scaling_test.go:124]: it simply validates the number of etcd members, which seems to be simple and enough, ideas?

File: test/e2e/vertical_scaling_test.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+	// step 1: add a new master node and wait until it is in Running state
+	machineName := createNewMasterMachine(ctx, t, machineClient)
+	ensureMasterMachineRunning(ctx, t, machineName, machineClient)
+
+	// step 2: wait until a new member shows up and check if it is healthy
+	ensureMembersCount(t, etcdClientFactory, 4)
+	memberName := machineNameToEtcdMemberName(ctx, t, kubeClient, machineClient, machineName)
+	ensureHealthyMember(t, etcdClientFactory, memberName)
+
+	// TODO: step 3: clean-up: until we can scale down
```
Comment (p0lyn0mial) [inline test/e2e/vertical_scaling_test.go:59]: once we have `0.b` implemented from the doc.

File: test/e2e/vertical_scaling_test.go [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+		machinePhase := pointer.StringDeref(machine.Status.Phase, "Unknown")
+		if machinePhase == "Running" {
+			machineToClone = &machine
+			break
+		}
+		t.Logf("%q machine is in unexpected %q state", machine.Name, machinePhase)
+	}
+
+	if machineToClone == nil {
+		t.Fatal("unable to find a master node to clone")
```
Comment (hasbro17) [inline test/e2e/vertical_scaling_test.go:77]:
```suggestion
		t.Fatal("unable to find a running master machine to clone")
```

**Review by hasbro17 — COMMENTED** [review]
/approve

Just a couple of nits but looks pretty good otherwise.

**Comment by p0lyn0mial** [comment]
@hasbro17 thanks for reviewing. I updated the PR, PTAL.

**Comment by deads2k** [comment]
how long does this test take to execute?  I'd like to see it added to the openshift/origin serial bucket to have coverage across all clouds and configurations.

**Comment by hasbro17** [comment]
> how long does this test take to execute?

Seems like 5m16s on the [last run](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-etcd-operator/732/pull-ci-openshift-cluster-etcd-operator-master-e2e-operator/1490985091894087680).

**Pull request merged**